src/dom/rect.js

```diff
@@ -180,6 +194,50 @@ export default class Rect {
 	}
 
 	/**
+	 * Returns a new rect, a part of the original rect, which is actually visible to the user,
+	 * e.g. an original rect cropped by parent element rects which have `overflow` set in CSS
+	 * other than `"visible"`.
+	 *
+	 * If there's no such visible rect, which is when the rect is obscured by one or many of
+	 * the ancestors, `null` is returned.
+	 *
+	 * @returns {module:utils/dom/rect~Rect|null} A visible rect instance or `null`, if there's none.
+	 */
+	getVisible() {
+		const obj = this._obj;
+		let visibleRect = this.clone();
+
+		// There's no ancestor to crop <body> with the overflow.
+		if ( obj != global.document.body ) {
+			let parent = obj.parentNode || obj.commonAncestorContainer;
+
+			// Check the ancestors all the way up to the <body>.
+			while ( parent && parent != global.document.body ) {
+				const parentOverflow = global.window.getComputedStyle( parent ).overflow;
```

Comment:
...and I was right. `window.getComputedStyle` is slow. Twice the performance after https://github.com/ckeditor/ckeditor5-utils/pull/149/commits/24e1d149b0953fa7e63ff92a43e0f6f0ac08da65.